src/Features/Core/Portable/EmbeddedLanguages/EmbeddedLanguageDetector.cs

```diff
@@ -0,0 +1,355 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Features.EmbeddedLanguages
+{
+    internal readonly struct EmbeddedLanguageDetector
```

Comment:
all this was extracted out and pulled up from abstract language detector.  It is mostly the same except that it does not try to do any work wrt to determining options for a particular language.